### PR TITLE
Entity Registry and Dalek Entity

### PR DESCRIPTION
**Pull request by CommandrMoose**
Work based on Craig's Weeping Angels mod.

### CODE REVIEW

File: common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                add(Attributes.ARMOR, 5.0D);
+    }
+
+    public void onAiServerStep() {
+
+    }
+
+    @Override
+    public void aiStep() {
+
```
Comment (Jeryn99) [inline common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java]: Where super?

File: common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return createMonsterAttributes().
+                add(Attributes.ATTACK_DAMAGE, 8D).
+                add(Attributes.MAX_HEALTH, 20D).
+                add(Attributes.KNOCKBACK_RESISTANCE, 0.5D).
+                add(Attributes.MOVEMENT_SPEED, 0.3D).
+                add(Attributes.ARMOR, 5.0D);
+    }
+
+    public void onAiServerStep() {
+
```
Comment (Jeryn99) [inline common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java]: Where super?

**Review by Jeryn99 — COMMENTED** [review]
Code is fine! But in future - something core to the mod such as a entity/model registry should be done separately from a feature

As not doing this approach will lead to git conflicts or block someone else from working as they will need to wait on this being merged or they take it upon themselves re-creating this code on their branch leading to the aforementioned conflicts - considering the Dalek code is also apart of this PR aswell

Granted, this is likely a temporary PR but you will either need to move all core code to the main branch or merge this as is now and continue your work from there

In future, we should follow be doing ``feature/dalek`` as branch names and committing registries to the base branch first

File: common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java (hunk 1 above)
Comment (CommandrMoose) [reply common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java]: Removed from commit.

File: common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java (hunk 2 above)
Comment (CommandrMoose) [reply common/src/main/java/who/whocraft/common/entity/hostile/AbstractDalek.java]: Removed from commit.

**Comment by Jeryn99** [comment]
Merged early due to the nature of the PR blocking other progress as it contains files required for development of entities - continue daleks on a new branch: ``git checkout -b feature/daleks``

**Pull request merged**